winrm/errors.go

```diff
@@ -0,0 +1,33 @@
+package winrm
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ServerError struct {
```

Comment:
It's not a return struct, it's just there to catalog errors that the winrm server can return and that we should return to the end-user (or not).

I do agree that using the `Client` as the relay for errors when running a command is a bad practice. We already have an `err` in the `Command`, this should be enough.

I'm going to see how I can rewrite the patch to eliminate this.